src/game.py

```diff
@@ -0,0 +1,284 @@
+            guesses, count, board = remembered.pop()
+            if count >= len(guesses):
+                continue
+            remembered.append((guesses, count + 1, board))
+            workspace = list(board)
+            pos, n = guesses[count]
+            workspace[pos] = n
+            guesses = self.deduce(workspace)
+            if guesses is None:
+                return (remembered, workspace)
+            remembered.append((guesses, 0, workspace))
+        return [], None
+
+    def deduce(self, board):
+        """Get a board that require to deduce the value following the
+         sudoku rules
+
+        Keyword arguments:
+        board -- Array e.g.: [[None, None,..., None, None]]
+        return  -- Array of tuples [(Position, Value)], the number of array items depends of number gasses found
+        """
+        while True:
+            stuck, guess, count = True, None, 0
+            allowed, needed = self.calculate_bits(board)
+            try:
+                board, allowed, stuck, guess, count = self.move_on_board(board, allowed, stuck, guess, count)
+            except:
+                pass
+            if not stuck:
+                allowed, needed = self.calculate_bits(board)
+            board, stuck, guess, count = self.move_on_board_for_axis(needed, allowed, board, guess, count, stuck)
+            if stuck:
+                return self.get_is_stuck(guess)
+
+    def get_is_stuck(self, guess):
+        """Ask about guess value
+
+        Keyword arguments:
+        guess -- inherited from deduce function
+        return  -- return an integer from 1 to 9
+        """
+        if guess is not None:
+            random.shuffle(guess)
+        return guess
+
+    def move_on_board_for_axis(self, needed, allowed, board, guess, count, stuck):
+        """Move on board for each axis and guess needed and allowed values
+
+        Keyword arguments:
+        needed -- Array of numbers needed to resolve puzzle
+        allowed -- Array of numbers allowed to resolve puzzle
+        board -- Array e.g.: [[None, 1,..., 8, 6]]
+        guess -- Array of random numbers from 1 to 9
+        count -- Count how many times pick the best option of guesses
+        stuck -- Verify if game stuck returning true or false
+        return  -- Tuple of (board, stuck, guess, count)
+        """
+        for axis in xrange(3):
+            try:
+                board, stuck, guess, count = self.move_on_x_position(needed, allowed, board, guess, count, axis, stuck)
+            except:
+                pass
+        return board, stuck, guess, count
+
+    def move_on_x_position(self, needed, allowed, board, guess, count, axis, stuck):
+        """Move on axis position for each axis x  and guess needed and allowed values
+
+        Keyword arguments:
+        needed -- Array of numbers needed to resolve puzzle
+        allowed -- Array of numbers allowed to resolve puzzle
+        board -- Array e.g.: [[None, 1,..., 8, 6]]
```

Comment:
we need that the documentation of each method and parameter has a description according the context of the program and also an example of the expected value